src/fluree/db/json_ld/shacl.cljc

```diff
@@ -541,6 +561,33 @@
         #?(:clj (Pattern/compile (or valid-flags 0))
            :cljs (js/RegExp. (or valid-flags ""))))))
 
+(defn resolve-path-type
+  "Associate each property path object with its path type in order to govern path flake resolution during validation."
+  [db path-pid]
+  (go-try
+    (if-let [path-flake (->> (<? (query-range/index-range db :spot = [path-pid]))
+                             (remove #(#{const/$xsd:anyURI} (flake/p %)))
+                             (first))]
+      (let [o (flake/o path-flake)
+            p (flake/p path-flake)]
+        (cond
```

Comment:
I think this should be replaced with a `case` statement on `p` since you only use the `=` predicate.

Reply:
I tried that but unfortunately `case` does not evaluate the test-constants, so for example
```clojure
(case 260
   const/$sh:inversePath :inverse
   260 :matches-here)
;; => :matches-here
```
I could change it to `condp` `=`, which is recommended by the [clojure style guide](https://github.com/bbatsov/clojure-style-guide#case), though it doesn't say why.

Reply:
In that case, I think you can use our [`case+` utility macro](https://github.com/fluree/db/blob/cc2d4bd72ddf67410bd2d1fa4515157a08815c42/src/fluree/db/util/core.cljc#L361-L382)

Reply:
The docstring of the `case+` macro says this: "Don't use this in CLJS if your dispatch values are :const". It recommends using `const-case` if you need to handle `:const` dispatch values in both CLJ and CLJS, but I can't find that anywhere.

The docstring confuses me, though, because it recommends using regular-old `case` if you have `:const` dispatch values in cljs, which I do, but it doesn't work with `:const` dispatch values in clj, which I need, so is the "Don't use in CLJS" injunction actually mean "you don't have to use this if you are in cljs-only territory, but it will work fine so don't worry about it".

Reply:
Wait, I found what I think it's referring to by `const-case`, that's actually `fluree.db.util.clj-const/case`. That's perfect, I'll use that and see if I can update the docstring to be more helpful.